### ISSUE TITLE
Type-check jsdoc-annotated property assignments

### ISSUE THREAD

**Issue by ghost** [microsoft/TypeScript#17216]
**TypeScript Version:**  nightly (2.5.0-dev.20170712)

**Code**

```ts
// @ts-check

/** @type {string} */
exports.x = 1;
```

**Expected behavior:**

Error.

**Actual behavior:**

No error.
And if I import `x` in another module, it is typed as `string`.

**Comment by sandersn** [comment]
I think this only makes sense for the `exports.x` special case that we support elsewhere. In general 

```js
var o = {}
/** @type {number} */
o.x = 1
/** @type {string} */
o.y = 'hi'
```

should instead be

```js
var o = (/** @type { {x: number, y: string} } */ {});
o.x = 1
o.y = 'hi'
```

**Comment by andy-ms** [comment]
We should warn about the unused `@type` in that case then, since the former seems more intuitive.

**Comment by vilic** [comment]
I am encountering this issue while trying to type check JS configuration files.

**Comment by RyanCavanaugh** [comment]
This errors in 6.0 now
